nuitka/OptionParsing.py

```diff
@@ -649,6 +649,16 @@
 
 debug_group = parser.add_option_group("Debug features")
 
+debug_group.add_option(
+    "--profile-optimization-time",
+    action="store_true",
+    dest="profile_optimization",
+    default=False,
+    help="""\
+Profile the time of the optimization and stores it in a file named 'callgrind'. Can be opened with qcachegrind. Defaults to off.
```

Comment:
Check out `run-valgrind.py`. I personally would be OK with the file being named more sensibly, like `main_filename.callgrind` derived from what we compile, and for us to launch Valgrind directly. Probably also something to move to `nuitka.utils.profiling` I guess, together with the context manager.

Reply:
It would be cool if the CI would show the callgrind results, imagine being able to tell from a downloaded callgrind file, what kind of improvements we have. Not sure if this can be GitHub though, since compiling SciPy per commit is not really an option. But maybe we could have a separate workflow to make these artifacts, I would look into adding that.